packages/codecs-core/src/bytes.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Concatenates an array of `Uint8Array`s into a single `Uint8Array`.
+ */
+export const mergeBytes = (bytesArr: Uint8Array[]): Uint8Array => {
+    const totalLength = bytesArr.reduce((total, arr) => total + arr.length, 0);
+    const result = new Uint8Array(totalLength);
+    let offset = 0;
+    bytesArr.forEach(arr => {
+        result.set(arr, offset);
+        offset += arr.length;
+    });
+    return result;
+};
+
+/**
+ * Pads a `Uint8Array` with zeroes to the specified length.
+ * If the array is longer than the specified length, it is returned as-is.
+ */
+export const padBytes = (bytes: Uint8Array, length: number): Uint8Array => {
+    if (bytes.length >= length) return bytes;
+    const paddedBytes = new Uint8Array(length).fill(0);
```

Comment:
Is this as performant as doing two `set()` operations with only the bytes needed?

```ts
const out = new Uint8Array(length);
out.set(bytes);
out.set(new Uint8Array(length - bytes.length).fill(0), bytes.length);
return out;
```

Reply:
I don't know but my guess would be no because it's instantiating an extra `Uint8Array`.